pkg/apis/nfd/v1alpha1/nodefeaturerule/rule.go

```diff
@@ -108,6 +108,41 @@ func Execute(r *nfdv1alpha1.Rule, features *nfdv1alpha1.Features) (RuleOutput, e
 	return ret, nil
 }
 
+// ExecuteGroupRule executes the GroupRule against a set of input features, and return true if the
+// rule matches.
+func ExecuteGroupRules(r *nfdv1alpha1.GroupRule, features *nfdv1alpha1.Features) (bool, error) {
+	matched := false
+	if len(r.MatchAny) > 0 {
+		// Logical OR over the matchAny matchers
+		for _, matcher := range r.MatchAny {
+			if isMatch, matches, err := evaluateMatchAnyElem(&matcher, features); err != nil {
+				return false, err
+			} else if isMatch {
+				matched = true
+				klog.V(4).InfoS("matchAny matched", "ruleName", r.Name, "matchedFeatures", utils.DelayedDumper(matches))
+				// there's no need to evaluate other matchers in MatchAny
+				// One match is enough for MatchAny
+				break
+			}
+		}
+		if !matched {
+			return false, nil
+		}
+	}
+
+	if len(r.MatchFeatures) > 0 {
+		if isMatch, _, err := evaluateFeatureMatcher(&r.MatchFeatures, features); err != nil {
+			return false, err
+		} else if !isMatch {
+			klog.V(2).InfoS("rule did not match", "ruleName", r.Name)
+			return false, nil
+		}
+	}
+
+	klog.V(2).InfoS("rule matched", "ruleName", r.Name)
+	return true, nil
```

Comment:
We have the `vars` and `varsTemplate` fields in the NFG type. We should handle (and test) those as well. Or alternatively drop them from the type definition